control/config.py

```diff
@@ -20,7 +20,40 @@
     'control.squeeze_time_response': None,
     'forced_response.return_x': False,
 }
-defaults = dict(_control_defaults)
+
+
+class DefaultDict(dict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __setitem__(self, key, value):
+        super().__setitem__(self._check_deprecation(key), value)
+
+    def __missing__(self, key):
+        repl = self._check_deprecation(key)
+        if self.__contains__(repl):
+            return self[repl]
+        else:
+            raise KeyError
+
+    def copy(self):
+        return DefaultDict(self)
+
+    def get(self, key, default=None):
+        return super().get(self._check_deprecation(key), default)
+
+    def _check_deprecation(self, key):
+        if self.__contains__(f"deprecated.{key}"):
+            repl = self[f"deprecated.{key}"]
+            warnings.warn(f"config.defaults['{key}'] has been renamed to "
```

Comment:
Again just to check you're aware: I used `DeprecationWarning` in some of my own code, and was surprised when there were no visible warnings; they're only shown in a specific mode (or modes?).  Are we intending this only for our own use in unit tests, or for the user?  IIRC, Pandas has a custom `UserWarning` for its deprecations, which are visible (and annoying!) in "standard" mode.

Reply:
Yes that was bugging me too, thanks.

I guess our user base is primarily using python-control in their own custom scripts, not through some 3rd party module importing control. `DeprecationWarning` is for the test suites of the latter category.

How about we introduce a `ControlDeprecationWarning` subclassed from `UserWarning`, which is shown by default but can be suppressed easily?

Reply:
I think this is a good idea.  As the [docs](https://docs.python.org/3/library/warnings.html) say, "Repetitions [...] are typically suppressed", so they're not *too* annoying.

Reply:
Using a `FutureWarning` should be a good resolution then.